testcases/kernel/device-drivers/uart/serialcheck.sh

```diff
@@ -0,0 +1,95 @@
+#! /bin/sh
+###############################################################################
+# Copyright (C) 2015 Texas Instruments Incorporated - http://www.ti.com/
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation version 2.
+#
+# This program is distributed "as is" WITHOUT ANY WARRANTY of any
+# kind, whether express or implied; without even the implied warranty
+# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+###############################################################################
+
+source "common.sh"  # Import do_cmd(), die() and other functions
+
+#### Functions definitions ####
+usage()
+{
+    echo "usage: ./${0##*/} [-r UART_RATE] [-l LOOPS] [-x to enable HW flow control]"
+    exit 1
+}
+
+check_requirements() {
+    which serialcheck || die "serialcheck binary is not available. Please install it"
+}
+
```

Comment:
So this depends on the serialcheck binary to be installed by a user from that github repository.

I wonder how to solve this, since we are usually trying to make things work out of the box for LTP.

Reply:
That's what @sw_requires metadata is used for.
There is no single answer here and this is an area where more standardization work is required. In our case, the rootfs comes w/ serialcheck already installed but that won't be the case for other rootfs.